hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/mor/MergeOnReadInputFormat.java

```diff
@@ -753,7 +758,7 @@ private Option<IndexedRecord> mergeRowWithLog(
         String curKey) throws IOException {
       final HoodieAvroRecord<?> record = (HoodieAvroRecord) scanner.getRecords().get(curKey);
       GenericRecord historyAvroRecord = (GenericRecord) rowDataToAvroConverter.convert(tableSchema, curRow);
-      return record.getData().combineAndGetUpdateValue(historyAvroRecord, tableSchema);
+      return record.getData().combineAndGetUpdateValue(historyAvroRecord, tableSchema, payloadConf);
```

Comment:
Nice catch, can we write a test case in `TestWriteMergeOnRead` ?

Reply:
The test case has been completed. and check whether there are areas for improvement. Thank you, `TestWriteMergeOnRead .testPrecombinFieldOnSnapshotQuery`